api/clients/v2/verification/cert_verifier.go

```diff
@@ -316,3 +309,26 @@ func (cv *CertVerifier) MaybeWaitForBlockNumber(ctx context.Context, targetBlock
 		}
 	}
 }
+
+// getVerifierCaller returns a ContractEigenDACertVerifierCaller that corresponds to the input certVerifierAddress
+//
+// This method caches ContractEigenDACertVerifierCaller instances, since their construction requires acquiring a lock
+// and parsing json, and is therefore not trivially inexpensive.
+func (cv *CertVerifier) getVerifierCaller(
+	certVerifierAddress string,
+) (*verifierBindings.ContractEigenDACertVerifierCaller, error) {
+
+	existingCaller, valueExists := cv.verifierCallers.Load(certVerifierAddress)
+	if valueExists {
+		return existingCaller.(*verifierBindings.ContractEigenDACertVerifierCaller), nil
```

Comment:
any value in instead safe casting the caster to the binding type? looking at this code it should be impossible but without an invariant enforcement this does present a hypothetical panic

Reply:
Good point, fixed in [74ec1b9e](https://github.com/Layr-Labs/eigenda/pull/1331/commits/74ec1b9ef482de085c86853cacde50fee2de1960)